components/browser/awesomebar/src/main/java/mozilla/components/browser/awesomebar/BrowserAwesomeBar.kt

```diff
@@ -95,12 +95,12 @@ class BrowserAwesomeBar @JvmOverloads constructor(
     @Synchronized
     override fun addProviders(vararg providers: AwesomeBar.SuggestionProvider) {
         providers.forEach { provider ->
-            val existingProvider = this.providers.find { it.id == provider.id }
-            existingProvider?.let {
+            if (containsProvider(provider)) {
                 throw IllegalStateException("Failed to add provider " +
                         "${provider.id} of type ${provider::class.java.name}. " +
-                        "Provider with the same ID already exists: ${it::class.java.name}")
+                        "Provider with the same ID already exists.")
```

Comment:
I think it's valuable to have the other providers type in the exception message. Theoretically, there could be two providers with the same ID but different types (or packages/versions) on the classpath. And that would be good to know as part of the exception.

My vote is to keep the original implementation of this method, and applications can use the new `containsProvider`.

Reply:
I agree. Will make the changes you suggested